### ISSUE TITLE
vdist returns incorrect answer when Lat1 and Lat2 are both zero or close to zero

### ISSUE THREAD

**Issue by bcminch** [scivision/pymap3d#29]
When checking a distance along the equator, vdist will return an incorrect result. By adjusting the latitudes so they are close to zero, vdist will correct this behavior.

The following code illustrates the issue:

```
from pymap3d.vincenty import vdist
print(0, '\t', vdist(0, 0, 0, 1)[0])
for i in range(16, -1, -1):
  try:
    print(1*10**-i, '\t', vdist(1*10**-i, 0, 1*10**-i, 1)[0])
  except ValueError as e:
    print(1*10**-i, '\t', f'Failed: {e.args}')
```

This code prints the following:

```
0 	 110946.29557670657
1e-16 	 110946.29557670657
1e-15 	 110946.29557670657
1e-14 	 110946.29557670657
1e-13 	 110946.29557670657
1e-12 	 110946.29557670657
1e-11 	 110946.29557670657
1e-10 	 110946.29557670657
1e-09 	 110946.29557670657
1e-08 	 110946.29557670657
1e-07 	 110946.29557670657
1e-06 	 Failed: ('math domain error',)
1e-05 	 111319.4907932247
0.0001 	 111319.49079305801
0.001 	 111319.49077638453
0.01 	 111319.48910904085
0.1 	 111319.32237470953
1 	 111302.64933938583
```

The values for 1e-05 and greater appear to be correct, and agree with the output from the Matlab distance function. The values for 0 and from 1e-16 through 1e-07 are all off by about the same amount, and 1e-06 produces a math domain error.

**Comment by scivision** [comment]
I took a look with current pymap3d `master` and i get for:

```python
for i in range(1, 16):
    lat = 10**(-i)
    dist_m, az_deg = vdist(lat, 0, -lat, 0)
    print(f"latitude {lat}: {dist_m} meters {az_deg} deg azimuth")
```

```
latitude 0.1: 22114.855389836863 meters 180.0 deg azimuth
latitude 0.01: 2211.4855166606058 meters 180.0 deg azimuth
latitude 0.001: 221.1485516437375 meters 180.0 deg azimuth
latitude 0.0001: 22.11485516435142 meters 180.0 deg azimuth
latitude 1e-05: 2.2114855164351206 meters 180.0 deg azimuth
latitude 1e-06: 0.22114855164351196 meters 180.0 deg azimuth
latitude 1e-07: 0.022114855164351195 meters 180.0 deg azimuth
latitude 1e-08: 0.0022114855164351195 meters 180.0 deg azimuth
latitude 1e-09: 0.00022114855164351196 meters 180.0 deg azimuth
latitude 1e-10: 2.2114855164351196e-05 meters 180.0 deg azimuth
latitude 1e-11: 2.2114855164351194e-06 meters 180.0 deg azimuth
latitude 1e-12: 2.2114855164351195e-07 meters 180.0 deg azimuth
latitude 1e-13: 2.2114855164351196e-08 meters 180.0 deg azimuth
latitude 1e-14: 2.2114855164351196e-09 meters 180.0 deg azimuth
latitude 1e-15: 2.2114855164351199e-10 meters 180.0 deg azimuth
```

**Comment by scivision** [comment]
This is related to #27 and fixed in 45324a7
Thanks again. I need to apply such checks to each function. I checked this vs. Matlab with https://github.com/scivision/pymap3d/blob/master/examples/vdist_stability.py
